katsdpimager/report.py

```diff
@@ -0,0 +1,356 @@
+        self.frequency_range = bokeh.models.Range1d(
+            self.common.frequencies[0].to_value(FREQUENCY_PLOT_UNIT),
+            self.common.frequencies[-1].to_value(FREQUENCY_PLOT_UNIT),
+            bounds='auto'
+        )
+        self.channel_range = bokeh.models.Range1d(0, self.common.channels - 1, bounds='auto')
+        self.time_on_target = katsdpimager.metadata.time_on_target(dataset, target)
+        self.predicted_natural_noise: Optional[u.Quantity] = None
+        if self.common.sefd is not None and len(self.common.antennas) > 1:
+            n = len(self.common.antennas)
+            # Correlator efficiency is already folded in to self.common.sefd
+            denom = math.sqrt(2 * n * (n - 1) * self.time_on_target * self.common.channel_width)
+            self.predicted_natural_noise = self.common.sefd / denom / u.beam
+
+        mask = katsdpimager.metadata.target_mask(dataset, target)
+        self.timestamps = dataset.timestamps[mask]
+        self.time_range = bokeh.models.Range1d(
+            datetime.fromtimestamp(self.timestamps[0], timezone.utc),
+            datetime.fromtimestamp(self.timestamps[-1], timezone.utc)
+        )
+        array_ant = dataset.sensor['Antennas/array/antenna'][0]
+        self.elevation = target.azel(timestamp=self.timestamps, antenna=array_ant)[1] << u.rad
+        self.parallactic_angle = target.parallactic_angle(
+            timestamp=self.timestamps, antenna=array_ant) << u.rad
+
+    @property
+    def name(self) -> str:
+        return self.target.name
+
+    @property
+    def description(self) -> str:
+        return self.target.description
+
+    def _add_channel_range(self, fig: bokeh.plotting.Figure) -> None:
+        fig.extra_x_ranges = {'channel': self.channel_range}
+        fig.add_layout(bokeh.models.LinearAxis(x_range_name='channel', axis_label='Channel'),
+                       'above')
+
+    def make_channel_data_source(self) -> bokeh.models.ColumnDataSource:
+        data = {
+            'frequency': self.common.frequencies.to_value(FREQUENCY_PLOT_UNIT),
+            'status': self.status,
+            'noise': self.noise.to_value(FLUX_PLOT_UNIT),
+            'peak': self.peak.to_value(FLUX_PLOT_UNIT)
+        }
+        if self.predicted_natural_noise is not None:
+            predicted_noise = self.predicted_natural_noise * self.normalized_noise
+            data['predicted_noise'] = predicted_noise.to_value(FLUX_PLOT_UNIT)
+        return bokeh.models.ColumnDataSource(data)
+
+    def make_time_data_source(self) -> bokeh.models.ColumnDataSource:
+        data = {
+            'time': [datetime.fromtimestamp(ts, timezone.utc) for ts in self.timestamps],
+            'elevation': self.elevation.to_value(ANGLE_PLOT_UNIT),
+            'parallactic_angle': self.parallactic_angle.to_value(ANGLE_PLOT_UNIT)
+        }
+        return bokeh.models.ColumnDataSource(data)
+
+    def make_plot_status(self, source: bokeh.models.ColumnDataSource) -> bokeh.model.Model:
```

Comment:
Looking at the example report you sent. When zooming in on the status plot using the 'wheel zoom' button it zooms both the x and y axes of the plot. Is there a way of changing the buttons on the plot so that the 'wheel zoom' function only zooms the x axis? Zooming in on the y axis in that figure seems pointless since it can only be one of 4 discrete values.
It might be possible, since the x axes are linked between the 'status' and 'flux density' plots (i.e. when you zoom on one the other changes just its x-axis). I got slightly lost in the bokeh API documentation trying to see if I could figure this out but came up empty myself.

Reply:
It looks like the tool can be set to only zoom one axis, and possibly also that the axis can be set to prevent zooming in (so that even using the rectangle zoom tool won't zoom in. I'll investigate.

Reply:
I've set the y axis so that it can't be zoomed. Unfortunately that seems to completely break the wheel zoom tool - probably a bug in Bokeh (there are other bugs related to wheel zoom not working in corner cases).